src/scenes/app/stores/index.js

```diff
@@ -1,65 +1,94 @@
 import { observable, action, toJS } from 'mobx';
+import ToDoController from '../../../networking/controllers/ToDoController';
 import strings from '../../../localization/en/strings';
 import Utils from '../../../helpers/Utils';
 
 class ObservableTodoStore {
   @observable items = [];
-  @observable error = '';
-  @observable start = true;
+  @observable error = null;
 
   @action
   clearAllDone = () => {
-    this.items = this.items.filter(element => element.completed === false);
+    const { generalError } = strings;
+    const itemsToRemove = this.getToDosDone();
+    itemsToRemove.forEach(async (toDoDone) => {
+      try {
+        const response = await ToDoController.deleteToDo(toDoDone.id);
+        this.removeToDo(response.data);
+      } catch (error) {
+        this.error = generalError;
```

Comment:
You're updating this observable outside of an action scope.

Reply:
we talk about that, my bad!